### PR TITLE
Auth/JWT

### PR DESCRIPTION
**Pull request by GigiDev7**
added logic on login service to sign and create jwt token, i left first argument empty, because i'll use user id for that when i'll have DB

### CODE REVIEW

**Comment by GigiDev7** [comment]
added middleware to protect routes

**Comment by GigiDev7** [comment]
in middleware when i decode data i'll have that user Id which i'll attach to req object and use it in the next middleware

**Comment by GigiDev7** [comment]
added registration, it's not in MVP but its somewhat same to login so not a big deal

File: middlewares/protectAuth.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const jwt = require("jsonwebtoken");
+
+const protectAuth = (req, res, next) => {
+  if (!req.headers.authorization) {
+    return res.status(401).json({ message: "Unauthorized" });
+  }
+
+  const token = req.headers.authorization.split(" ")[1];
+  jwt.verify(token, process.env.JWT_SECRET, (err, decodedData) => {
```
Comment (Lokki17) [inline middlewares/protectAuth.js:9]: what if req.headers.authorization.split(" ")[1] - undefined? (just asking)

File: services/user.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  const token = createToken(/* userId*/);
+
+  return { token /* user */ };
+};
+
+const registerUser = async (userData) => {
+  //check if emails is not registered already in DB
+
+  const token = createToken(/* userId*/);
+  return { token /* user */ };
```
Comment (Lokki17) [inline services/user.js]: Why you return token on register? Register should create new user. Then user should use loginUser to get token

**Review by Lokki17 — CHANGES_REQUESTED** [review]
Fix remark and merge

File: middlewares/protectAuth.js (hunk 1 above)
Comment (GigiDev7) [reply middlewares/protectAuth.js:9]: ill add that in the first if check

File: services/user.js (hunk 2 above)
Comment (GigiDev7) [reply services/user.js]: in loginUser i have different functionality, or the functionality that i dont use on registering. like when i log in i check in DB if i have that user than i have to compare passwords create token and return it to make user logged in, while when user registers, i create new user document in mongo collection i encrypt password and then also create token and return it to make user logged in. am i not correct in that logic?

Comment (GigiDev7) [reply services/user.js]: i use registerUser function on api/user/register route and loginUser on api/user/login route

Comment (GigiDev7) [reply services/user.js]: you mean that when user fills the registration form and presses the 'register' button then user should go to login page fill the login form to login?

**Comment by GigiDev7** [comment]
so as i understood on registretion user will not be logged in , so i removed that logic from registerUser function

File: middlewares/protectAuth.js (hunk 1 above)
Comment (Lokki17) [reply middlewares/protectAuth.js:9]: Maybe jwt.verify can handle this case, I dont know. But it's ok now

**Review by Lokki17 — APPROVED** [review]

File: services/user.js (hunk 2 above)
Comment (Lokki17) [reply services/user.js]: `you mean that when user fills the registration form and presses the 'register' button then user should go to login page fill the login form to login`
Yes. Register endpoint should only create user and login endpoint should only generate token (single responsibility principle)

**Pull request merged**